src/rdb/rdb_tx.c

```diff
@@ -905,7 +958,38 @@ rdb_tx_apply(struct rdb *db, uint64_t index, const void *buf, size_t len,
 			return rc;
 		}
 		p += n;
-		crit = hdr.critical;
+		crit = (hdr.flag & RDB_TX_HDR_FLAG_CRIT);
+
+		if ((hdr.flag & RDB_TX_HDR_FLAG_COMPACT) && !raft_is_leader(db->d_raft)) {
+			uint64_t	idx;
+			//ABT_mutex_lock(db->d_raft_mutex);
```

Comment:
Right, IIRC, `rdb_tx_apply` is called by the `log_offer` callback. So we shall have already acquired `d_raft_mutex`.